src/Features/Core/Portable/ChangeSignature/DelegateInvokeMethodReferenceFinder.cs

```diff
@@ -90,7 +90,7 @@ protected override bool CanFind(IMethodSymbol symbol)
                         if (convertedType != null)
                         {
                             convertedType =
-                                SymbolFinder.FindSourceDefinitionAsync(convertedType, document.Project.Solution, cancellationToken).WaitAndGetResult(cancellationToken)
+                                SymbolFinder.FindSourceDefinitionAsync(convertedType, document.Project.Solution, cancellationToken).WaitAndGetResult_CanCallOnBackground(cancellationToken)
```

Comment:
why?

Reply:
we're in an async method... why is this synchronously blocking?

Reply:
➡️ This really needs a `WhereAsync` extension or something similar. Either way, this pull request didn't make the situation any worse than it already was, and the correction for this falls under the VSTHRD002 (avoid problematic synchronous waits) umbrella which will soon have analyzer enforcement.